src/superstring/core.clj

```
@@ -713,3 +713,173 @@
+          \Ç          "%C7"
+          \È          "%C8"
+          \É          "%C9"
+          \Ê          "%CA"
+          \Ë          "%CB"
+          \Ì          "%CC"
+          \Í          "%CD"
+          \Î          "%CE"
+          \Ï          "%CF"
+          \Ð          "%D0"
+          \Ñ          "%D1"
+          \Ò          "%D2"
+          \Ó          "%D3"
+          \Ô          "%D4"
+          \Õ          "%D5"
+          \Ö          "%D6"
+          \Ø          "%D8"
+          \Ù          "%D9"
+          \Ú          "%DA"
+          \Û          "%DB"
+          \Ü          "%DC"
+          \Ý          "%DD"
+          \Þ          "%DE"
+          \ß          "%DF"
+          \à          "%E0"
+          \á          "%E1"
+          \â          "%E2"
+          \ã          "%E3"
+          \ä          "%E4"
+          \å          "%E5"
+          \æ          "%E6"
+          \ç          "%E7"
+          \è          "%E8"
+          \é          "%E9"
+          \ê          "%EA"
+          \ë          "%EB"
+          \ì          "%EC"
+          \í          "%ED"
+          \î          "%EE"
+          \ï          "%EF"
+          \ð          "%F0"
+          \ñ          "%F1"
+          \ò          "%F2"
+          \ó          "%F3"
+          \ô          "%F4"
+          \õ          "%F5"
+          \ö          "%F6"
+          \÷          "%F7"
+          \ø          "%F8"
+          \ù          "%F9"
+          \ú          "%FA"
+          \û          "%FB"
+          \ü          "%FC"
+          \ý          "%FD"
+          \þ          "%FE"
+          \ÿ          "%FF"}))
+
+(defn url-encode
+  "Returns the s string URL encoded. If s is nil, nil is returned.
+  It accepts a :strict? flag (false by default) to encode only URL reserved
+  characters as per https://tools.ietf.org/html/rfc3986#section-2.2."
+  ^String [^String s & {:keys [strict?]}]
+  {:pre   [(or (nil? s) (string? s))]
+   :post  [(or (nil? %) (string? %))]}
+  (and s
```

Calling string manipulation functions on `nil` doesn't make any sense to me, so in the spirit of fail fast I'd prefer asserting `string?` in `:pre`.  This is what's been done elsewhere in superstring.